### PR TITLE
REAME and CONTRIBUTING updates

### PR DESCRIPTION
**Pull request by l0rd**
### What does this PR do?

- Add information about what DWO does and how to use it in README.md
- Move configuration (dwoc) documentation under `docs/dwo-configuration.md`
- Move development documentation in `CONTRIBUTING.md`

### What issues does this PR fix or reference?

https://github.com/devfile/devworkspace-operator/issues/1334

### Is it tested? How?

The instructions in the new section have been tested on developer sandbox. 

### PR Checklist

- [ ] E2E tests pass (when PR is ready, comment `/test v8-devworkspace-operator-e2e, v8-che-happy-path` to trigger)
    - [ ] `v8-devworkspace-operator-e2e`: DevWorkspace e2e test
    - [ ] `v8-che-happy-path`: Happy path for verification integration with Che

### CODE REVIEW

**Review by ibuziuk — COMMENTED** [review]
Great contribution, thank you!

**Review by ibuziuk — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,104 +1,80 @@
...
 
-> Note: The operator requires internet access from containers to work. By default, `crc setup` may not provision this, so it's necessary to configure DNS for Docker:
-> ```
-> # /etc/docker/daemon.json
-> {
->   "dns": ["192.168.0.1"]
-> }
-> ```
+If the DevWorkspace Operator is not already available amongst the `PackageManifests` (use command
+`kubectl get packagemanifest -n olm | grep devworkspace` to check it) you add should add a `CatalogSource` in the
```
Comment (dkwon17) [inline README.md:73]:
```suggestion
`kubectl get packagemanifest -n olm | grep devworkspace` to check it) you should add a `CatalogSource` in the
```

**Review by dkwon17 — APPROVED** [review]

**Comment by dkwon17** [comment]
@l0rd sorry, could you add the fix in this commit I've added: 12b527f48e1fabd9444d62079c39c46e45c9e67b into a signed-off commit? I will merge this PR right after that.

**Comment by l0rd** [comment]
> @l0rd sorry, could you add the fix in this commit I've added: [12b527f](https://github.com/devfile/devworkspace-operator/commit/12b527f48e1fabd9444d62079c39c46e45c9e67b) into a signed-off commit? I will merge this PR right after that.

Thank you @dkwon17. I have included your change in the last commit of this PR.

**Review by dkwon17 — APPROVED** [review]

**Pull request merged**